File: comparison/results/haiku/baseline/humaneval/HumanEval_129/java/solution.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static ArrayList<Long> minPath(ArrayList<ArrayList<Long>> grid, long k) {
        int n = grid.size();
        
        // Find the position of value 1
        int row = -1, col = -1;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (grid.get(i).get(j) == 1) {
                    row = i;
                    col = j;
                    break;
                }
            }
            if (row != -1) break;
        }
        
        ArrayList<Long> path = new ArrayList<>();
        path.add(grid.get(row).get(col));
        
        // Build the path greedily
        for (long step = 1; step < k; step++) {
            int[][] directions = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
            long minValue = Long.MAX_VALUE;
            int nextRow = -1, nextCol = -1;
            
            for (int[] dir : directions) {
                int newRow = row + dir[0];
                int newCol = col + dir[1];
                if (newRow >= 0 && newRow < n && newCol >= 0 && newCol < n) {
                    long value = grid.get(newRow).get(newCol);
                    if (value < minValue) {
                        minValue = value;
                        nextRow = newRow;
                        nextCol = newCol;
                    }
                }
            }
            
            if (nextRow != -1) {
                path.add(grid.get(nextRow).get(nextCol));
                row = nextRow;
                col = nextCol;
            }
        }
        
        return path;
    }
}

File: comparison/results/haiku/baseline/humaneval/HumanEval_129/javascript/solution.js
function minPath(grid, k) {
    const n = grid.length;
    
    // Find position of value 1
    let startRow = -1, startCol = -1;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (grid[i][j] === 1) {
                startRow = i;
                startCol = j;
                break;
            }
        }
        if (startRow !== -1) break;
    }
    
    // Find minimum neighbor value of cell 1
    let minNeighbor = Infinity;
    const directions = [[-1, 0], [1, 0], [0, -1], [0, 1]];
    for (const [dr, dc] of directions) {
        const nr = startRow + dr;
        const nc = startCol + dc;
        if (nr >= 0 && nr < n && nc >= 0 && nc < n) {
            minNeighbor = Math.min(minNeighbor, grid[nr][nc]);
        }
    }
    
    // Build lexicographically smallest path by alternating between 1 and minNeighbor
    const path = [1];
    for (let i = 1; i < k; i++) {
        if (i % 2 === 1) {
            path.push(minNeighbor);
        } else {
            path.push(1);
        }
    }
    
    return path;
}
